captum/_utils/gradient.py

```diff
@@ -251,6 +258,8 @@ def forward_hook(module, inp, out=None):
 
             if not is_eval_tuple:
                 eval_tsrs = (eval_tsrs,)
+            if require_layer_grads:
+                apply_gradient_requirements(eval_tsrs, warn=False)
```

Comment:
Default behavior now warn=False too, right ? Perhaps we don't want that warn=False is default behavior.